applications/vanilla/views/categories/helper_functions.php

```diff
@@ -54,46 +54,34 @@ function categoryString($Rows) {
     }
 endif;
 
-if (!function_exists('GetOptions')):
+if (!function_exists('getOptions')):
     /**
-     * Render options that the user has for this category.
+     * Render options that the user has for this category. Returns an empty string if the session isn't valid.
+     *
+     * @param $category The category to render the options for.
+     * @return DropdownModule|string A dropdown with the category options or an empty string if the session is not valid.
+     * @throws Exception
      */
-    function getOptions($Category) {
-        if (!Gdn::session()->isValid())
-            return;
-
-        $Sender = Gdn::controller();
-
-
-        $Result = '';
-        $Options = '';
-        $CategoryID = val('CategoryID', $Category);
-
-        $Result = '<div class="Options">';
-        $TKey = urlencode(Gdn::session()->TransientKey());
+    function getOptions($category) {
+        if (!Gdn::session()->isValid()) {
+            return '';
+        }
+        $sender = Gdn::controller();
+        $categoryID = val('CategoryID', $category);
 
-        // Mark category read.
-        $Options .= '<li rel="MarkRead">'.anchor(t('Mark Read'), "/category/markread?categoryid=$CategoryID&tkey=$TKey").'</li>';
+        $dropdown = new DropdownModule();
+        $tk = urlencode(Gdn::session()->TransientKey());
+        $hide = (int)!val('Following', $category);
 
-        // Follow/Unfollow category.
-        if (!val('Following', $Category))
-            $Options .= '<li rel="Hide">'.anchor(t('Unhide'), "/category/follow?categoryid=$CategoryID&value=1&tkey=$TKey").'</li>';
-        else
-            $Options .= '<li rel="Hide">'.anchor(t('Hide'), "/category/follow?categoryid=$CategoryID&value=0&tkey=$TKey").'</li>';
+        $dropdown->addLink(t('Mark Read'), '/category/markread?categoryid='.$categoryID.'&tkey='.$tk, 'mark-read');
+        $dropdown->addLink(t($hide ? 'Unhide' : 'Hide'), '/category/follow?categoryid='.$categoryID.'&value='.$hide.'&tkey='.$tk, 'hide');
 
         // Allow plugins to add options
-        $Sender->EventArguments['Options'] = &$Options;
-        $Sender->fireEvent('CategoryOptions');
-
```

Comment:
Removing this potentially breaks backwards compatibility. However, the event isn't being used in Vanilla at all. To maintain backwards compatibility, It'd take some parsing of the options string in order to break it down into a dropdown element. This isn't something I want to devote time to, as I'm skeptical that it's even being used.